airbyte-integrations/connectors/source-linkedin-ads/source_linkedin_ads/components.py

```diff
@@ -0,0 +1,362 @@
+            authenticator=self._authenticator,
+            use_cache=self.use_cache,
+            backoff_strategy=backoff_strategies,
+            disable_retries=self.disable_retries,
+            message_repository=self.message_repository,
+        )
+
+
+@dataclass
+class AnalyticsDatetimeBasedCursor(DatetimeBasedCursor):
+    @staticmethod
+    def chunk_analytics_fields(
+        fields: List = ANALYTICS_FIELDS_V2,
+        fields_chunk_size: int = FIELDS_CHUNK_SIZE,
+    ) -> Iterable[List]:
+        """
+        Chunks the list of available fields into the chunks of equal size.
+        """
+        # Make chunks
+        chunks = list((fields[f : f + fields_chunk_size] for f in range(0, len(fields), fields_chunk_size)))
+        # Make sure base_fields are within the chunks
+        for chunk in chunks:
+            if "dateRange" not in chunk:
+                chunk.append("dateRange")
+            if "pivotValues" not in chunk:
+                chunk.append("pivotValues")
+        yield from chunks
+
+    def _partition_daterange(
+        self, start: datetime.datetime, end: datetime.datetime, step: Union[datetime.timedelta, Duration]
+    ) -> List[StreamSlice]:
+        start_field = self._partition_field_start.eval(self.config)
+        end_field = self._partition_field_end.eval(self.config)
+        dates = []
+        while start <= end:
+            next_start = self._evaluate_next_start_date_safely(start, step)
+            end_date = self._get_date(next_start - self._cursor_granularity, end, min)
+            date_slice_with_fields: List = []
+            for fields_set in self.chunk_analytics_fields():
+                date_range = {
+                    "start.day": start.day,
+                    "start.month": start.month,
+                    "start.year": start.year,
+                    "end.day": end_date.day,
+                    "end.month": end_date.month,
+                    "end.year": end_date.year,
+                }
+
+                fields = ",".join(fields_set)
+                date_slice_with_fields.append(
+                    {
+                        start_field: self._format_datetime(start),
+                        end_field: self._format_datetime(end_date),
+                        "fields": fields,
+                        **date_range,
+                    }
+                )
+            dates.append(StreamSlice(partition={}, cursor_slice={"field_date_chunks": date_slice_with_fields}))
+            start = next_start
+        return dates
+
+
+@dataclass
+class LinkedInAdsRecordExtractor(RecordExtractor):
+    def _date_time_to_rfc3339(self, record: MutableMapping[str, Any]) -> MutableMapping[str, Any]:
+        """
+        Transform 'date-time' items to RFC3339 format
+        """
+        for item in record:
+            if item in ["lastModified", "created"] and record[item]:
+                record[item] = pendulum.parse(record[item]).to_rfc3339_string()
```

Comment:
nit, just to speed up
```suggestion
        for item in ["lastModified", "created"]:
            if record.get(item):
                record[item] = pendulum.parse(record[item]).to_rfc3339_string()
```

Reply:
thnx 
updated